### ISSUE TITLE
Uncaught TypeError: Cannot read property 'type' of undefined

### ISSUE THREAD

**Issue by ChrisRast** [gandm/language-babel#381]
Hi,
No idea on how to reproduce this issue, but I think it happened (twice) after I focused back in a pane (I work with two pane, left-right).

My file contained a ES6 React class with JSX.

**Atom**: 1.17.2 x64
**Electron**: 1.3.15
**OS**: Mac OS X 10.10.5
**Thrown From**: [language-babel](https://github.com/gandm/language-babel) package 2.65.1

### Stack Trace

Uncaught TypeError: Cannot read property 'type' of undefined

```
At /Users/esl-christophe/.atom/packages/language-babel/lib/auto-indent.coffee:547

TypeError: Cannot read property 'type' of undefined
    at AutoIndent.module.exports.AutoIndent.indentUntokenisedLine (/packages/language-babel/lib/auto-indent.coffee:547:17)
    at AutoIndent.module.exports.AutoIndent.indentJSX (/packages/language-babel/lib/auto-indent.coffee:538:14)
    at AutoIndent.module.exports.AutoIndent.changedCursorPosition (/packages/language-babel/lib/auto-indent.coffee:110:6)
    at AutoIndent.changedCursorPosition (/packages/language-babel/lib/auto-indent.coffee:1:1)
    at /packages/language-babel/lib/auto-indent.coffee:70:68
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at TextEditor.module.exports.TextEditor.cursorMoved (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:2136:1)
    at Selection.module.exports.Selection.markerDidChange (/Applications/Atom.app/Contents/Resources/app/src/selection.js:921:1)
    at /Applications/Atom.app/Contents/Resources/app/src/selection.js:41:1
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at DisplayMarker.module.exports.DisplayMarker.notifyObservers (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:248:1)
    at /Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:53:1
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at Marker.module.exports.Marker.emitChangeEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/marker.js:421:1)
    at Marker.module.exports.Marker.update (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/marker.js:362:1)
    at Marker.module.exports.Marker.setHeadPosition (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/marker.js:144:1)
    at DisplayMarker.module.exports.DisplayMarker.setHeadBufferPosition (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:146:1)
    at DisplayMarker.module.exports.DisplayMarker.setHeadScreenPosition (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:154:1)
    at /Applications/Atom.app/Contents/Resources/app/src/cursor.js:72:1
    at Cursor.module.exports.Cursor.changePosition (/Applications/Atom.app/Contents/Resources/app/src/cursor.js:684:1)
    at Cursor.module.exports.Cursor.setScreenPosition (/Applications/Atom.app/Contents/Resources/app/src/cursor.js:70:1)
    at /Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1907:1
    at TextEditor.module.exports.TextEditor.moveCursors (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:2130:1)
    at TextEditor.module.exports.TextEditor.setCursorScreenPosition (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1906:1)
    at TextEditorComponent.module.exports.TextEditorComponent.onMouseDown (/Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:733:1)
    at HTMLDivElement.<anonymous> (/Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:3:1)
```

### Commands

```
     -2:42.4.0 editor:duplicate-lines (input.hidden-input)
     -2:40.6.0 find-and-replace:select-next (input.hidden-input)
     -2:36.7.0 core:save (input.hidden-input)
     -1:05.3.0 core:move-down (input.hidden-input)
     -1:04.7.0 autocomplete-plus:confirm (input.hidden-input)
     -1:04.4.0 core:save (input.hidden-input)
  7x -0:10.5.0 core:backspace (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.29.26 
atom-handlebars 1.3.0 
atom-html-preview 0.1.22 
atom-spotified 0.8.1 
atom-spotify2 2.1.2 
autoclose-html 0.23.0 
autocomplete-html-entities 0.1.0 
autocomplete-modules 1.6.10 
busy-signal 1.4.3 
clipboard-history 0.6.6 
color-picker 2.2.5 
compare-files 0.8.0 
docblockr 0.10.3 
editorconfig 2.2.2 
es6-javascript 1.0.0 
escape-utils 0.5.3 
file-icons 2.1.7 
file-type-icons 1.3.4 
file-types 0.5.5 
genesis-syntax 1.0.9 
genesis-ui 0.5.0 
git-plus 7.9.3 
gl-light-syntax 2.2.0 
highlight-line 0.12.0 
hyperclick 0.0.40 
intentions 1.1.2 
js-hyperclick 1.11.0 
language-babel 2.65.1 
language-diff 0.7.0 
language-htaccess 0.1.1 
language-markdown 0.23.0 
language-nunjucks 4.1.0 
linter 2.1.4 
linter-eslint 8.2.1 
linter-js-yaml 1.2.7 
linter-php 1.3.2 
linter-remark 4.0.1 
linter-sass-lint 1.7.4 
linter-ui-default 1.6.1 
markdown-preview-plus 2.4.9 
markdown-toc 0.4.2 
minimap 4.28.2 
php-hyperclick 1.2.1 
php-twig 4.0.0 
pigments 0.39.1 
project-manager 3.3.5 
sass-autocompile 0.13.3 
source-preview 0.5.2 
svg-preview 0.11.0 
todo-show 1.11.0 
```

Let me know If you need any info, I'm eager to help. =)
Cheers, 
Chris

**Comment by gandm** [comment]
Thanks for the report. I will take a look. One question - When working in split panes was it the same doc and were the cursors in both docs inside the same JSX block?

**Comment by ChrisRast** [comment]
No for both, two different files, with two different content.
If I notice another iteration of this issue I'll try to complete my report.

**Comment by gandm** [comment]
I think this is fixed in 2.66.1.
